src/AltinnCore/Runtime/RestControllers/ProcessController.cs

```diff
@@ -41,29 +66,445 @@ public class ProcessController : ControllerBase
+        }
+
         /// <summary>
         /// Gets a list of the next process elements that can be reached from the current process element.
         /// </summary>
         /// <returns>list of next process elements (tasks or events)</returns>
         [HttpGet("next")]
-        public async Task<ActionResult> GetNextProcessElements(
+        public async Task<ActionResult> GetNextElements(
+            [FromRoute] string org,
+            [FromRoute] string app,
+            [FromRoute] int instanceOwnerId,
+            [FromRoute] Guid instanceGuid)
+        {
+            Instance instance = await instanceService.GetInstance(app, org, instanceOwnerId, instanceGuid);
+            if (instance == null)
+            {
+                return NotFound();
+            }
+
+            ProcessKeeper = BpmnReader.Create(processService.GetProcessDefinition(org, app));
+
+            if (instance.Process == null)
+            {
+                return Conflict($"Process is not started. Use start!");
+            }
+
+            string currentTaskId = instance.Process.CurrentTask?.ElementId;
+
+            if (currentTaskId == null)
+            {
+                return Conflict($"Instance does not have valid info about currentTask");
+            }
+
+            List<string> nextElementIds = ProcessKeeper.NextElements(currentTaskId);
+
+            if (nextElementIds.Count == 0)
+            {
+                return NotFound("Cannot find any valid process elements that can be reached from current task");
+            }
+
+            return Ok(nextElementIds);           
+        }
+
+        /// <summary>
+        /// Change the instance's process state to next process element in accordance with process flow.
+        /// </summary>
+        /// <returns>new process state</returns>
+        [HttpPut("next")]
+        public async Task<ActionResult<ProcessState>> NextElement(
+            [FromRoute] string org,
+            [FromRoute] string app,
+            [FromRoute] int instanceOwnerId,
+            [FromRoute] Guid instanceGuid,
+            [FromQuery] string elementId)
+        {
+            Instance instance = await instanceService.GetInstance(app, org, instanceOwnerId, instanceGuid);
+            if (instance == null)
+            {
+                return NotFound("Cannot find instance!");
+            }
+
+            if (instance.Process == null)
+            {
+                return Conflict($"Process is not started. Use start!");
+            }
+
+            ProcessKeeper = BpmnReader.Create(processService.GetProcessDefinition(org, app));
+
+            string currentElementId = instance.Process.CurrentTask?.ElementId;
+
+            if (currentElementId == null)
+            {
+                return Conflict($"Instance does not have current task information!");
+            }            
+
+            string nextElement = CheckOrGetValidNextElement(currentElementId, elementId, out ActionResult nextElementError);
```

Comment:
elementId might be null as it is a query parameter.. will this handle getting null as input or will it break?